### DISCUSSION TITLE
[Design/Science] R&D rework

### DISCUSSION THREAD

**Discussion by AffectedArc07** [ParadiseSS13/Paradise#17756] [Documents]
Quick points to alleviate any confusion:
- This is not techwebs
- We are not getting departmental lathes
- The destructive analyser is going
- The experimentor is going

I will also point out this idea isn't entirely my own, but inspired by suggestions from other people.

# The problem

Currently, R&D is in a bad state. It is now _expected_ to be done within 20 minutes, which not only leaves scientists bored for the other 1 hour 40 minutes, but develops into an incredibly toxic gameplay of "wow R&D isn't done you're bad" or LOOCs of "use this (insert link to powergamer R&D tool here) to do R&D already". *There is also the entire issue of the EXPERIMENTOR but I will get to that can of worms later.*

Techwebs is often seen as a solution to this, but it itself is flawed, with people dictating "This is the most optimal solution, you will not deviate from it", and the UI code for it is clunky and laggy. A fair few people also dislike it, so I propose a solution.

# The solution

Hybrid R&D, Tycoon R&D, whatever you want to call it. The basic premise is that you earn research points over time. This can be done in many ways, including:
- Research computation servers (These must be kept cool if you want them to work quickly)
    - These can be built and generate points passively. They have a flat baseline, but can be overclocked for more performance. Obviously they use a lot of power. Possible ideas include a modular system where better components will generate more levels or do it more efficiently or similar.
- Research chemical decomposers (Create and insert a specific complex chem to gain research points from it)
    - Chem complexity is calculated by recipe steps. The more of the chem you make, the more diminished the returns get. 
    - (For the code nerds) The chem type and usage will be tracked globally so you cant cheese it
- Research explosion generator (Generates points from hitting close to the target)
    - Diminishing returns as you go on to avoid mass bombing. Also simulates only with toxins tanks. This allows you to waste fewer TTVs, as well as removes the chem cheese from toxins. 
    - This would only be able to be completed once, due to the ability to keep making the same mix since it doesnt deplete when calculating.
- Research slime core decomposer (Same as the chem thing but for xenobio)
    - Will be calculated the same as the chem complexit, or just with a picked list of far-bred slimes. Diminishing returns will happen when you use more slime cores
- Using the object analyser to discover strange objects (See below)
    - Strange objects will have a choice of being able to destroy a strange object for many points, or keep it for a few points. This it to avoid an obnoxious amount of strange objects in circulation.

Once you gain points, you can then spend them on levels. Materials 1 --> Materials 2 may cost 2500 points or so, with 2 --> 3 costing more than that. It will scale logarithmically, and ill likely ask @hal9000PR for a good scaling equation for it.

The destructive analyser and EXPERIMENTOR will be removed, with the EXPERIMENTOR room becoming free space to build servers or other point gain apparatus. The DA will be replaced with the "Object analyser", which can non-destructively analyse objects. This will be used to both discover strange objects and grant points based on what they do (Bluespace teleportation ones will grant more points than animal spawn ones, and this allows scientists to get a rough idea on what it might be as well), and it will also be used to allow for illegal technology research (You can analyse an emag to allow research up to illegal 7, but you will still have to put the points in).

The goal of this system is to slow R&D down and make it take more than 20 minutes so science isn't a powerhouse all round, but not resort to techwebs, while still giving R&D a choice of research paths. If its a calm round, they can put points into bio research and sort medbay out. If its war ops, they can pump points straight into hyperguns and mechs. 

As for free golems (and theta station) as Denth pointed out, I have a solution for those too. They can be given a separate R&D compute server, and strange objects could be dug out of rocks or something similar, allowing them to still do R&D things without a DA. R&D will also be split into separate networks, with the server controller getting a revamp because its awful at the moment. 

If anyone has any questions / feedback on this I will gladly take it 

--- 

Leadership verdicts:

Crazylemon: ☑️
Denth: ☑️
Doc: ☑️
Fox: ☑️
Mochi: ☑️
Neca: ☑️

**Comment by S34NW** [comment]
Question:
With the proposed removal of EXPERIMENTOR, how will the sometimes dangerous side effects of strange objects be tied into the object analyser? One of the main reasons EXPERIMENTOR is in a blast-proof room is due to its tendency to spew angry mobs (something I think should stay, science can risk angry mobs if they want more tech)

**Reply by AffectedArc07** [reply]
This is partly why I wanted to add the points "range" thing so people can guess what its going to do. I could add an "object danger" scale, and if its in the red, people can use this room
![image](https://user-images.githubusercontent.com/25063394/167399295-252934db-e845-4a81-bfd0-af7ba7c7e7a1.png)

Bonus points since they can then load the room with CO2 to clear out any mobs inside, then drain it after.

**Comment by Miraviel** [comment]
I love this! A little too draft-y for me to imagine, but here are my balance concerns for future consideration:

####  Research chemical decomposers (Create and insert a specific complex chem to gain research points from it)

I suggest adding a hardcap+cooldown(?) to this, I personally know a great many chem recipes by muscle memory and could just mass produce stuff in 10 minutes, reaching the same maxed out RND issue as we have now.

####  Research explosion generator (Generates points from hitting close to the target)

Another hard cap/cooldown request for this, lest A) we get cargo bombarded with tank transfer valve requests, replacing the current "order muh prototypes rn" or B) the station will shake non-stop from potawater bombs for them points, greatly disrupting the flow of the round.

#### Research slime core decomposer (Same as the chem thing but for xenobio)

Once per extract, right? Xenobio can be spammed.

#### In general

I wish to see this ingame but I have some concerns here:

**Spamming:**

If chems/explosions give a flat point bonus, they _will_ be spammed. Players want instant gratification, points over time is great, but it will be only relied on if they can overclock it to the point we get back to the level 7 by 20 minutes gameplay. I know this is the intention, but be ready for players trying to cheese it.
- Let's say you cap the maximum minute/possible points. Is the new meta intended to be overclocking the RND server and then spamming chem/explosions? I am trying to think about the current, "expected" behaviour - that is, to get stuff as fast as possible. Will 5-6 scientist just rush scichem, trying to make chems + potawater bombs and shake the station constantly?

**Short round antags (nukies, wizard, blob):**

This will greatly weaken science for nukie rounds (a good thing), but it will also inadvertently weaken cargo too (not sure if a good thing). Currently, during nukie rounds, RND gets done in about 8-10 minutes, ships the disks, cargo orders a lot of guns. If RND cannot get done anymore this fast, I suggest giving cargo a flat number of points from CC when war is declared to cut out science from this equation. Or a bump to RND levels from CC to make them able to do disks before the nukies arrive. Not enough to spam weapons yet but enough to get disks done/some equipment to be printed (I'm mainly thinking about medical stuff like cloner, sleeper, defibs, analyzers, handheld crew monitors...).

- Same as above, but for wizard. Do we want science to be not relevant during wizard rounds? (Not having an issue with it, I just toss it up here.)
- Same as above, but for roundstart blob. I know it is due for removal(?) but as long as we have it, we need to consider it a powerful early round antag.

**Roboticists and exosuits:**

Without levels, I am unsure what robotics can do apart from cleanbots, cyborgs, and ripleys. Can they still do their fun projects or are they locked behind levels?

**Reply by AffectedArc07** [reply]
> I suggest adding a hardcap+cooldown(?) to this, I personally know a great many chem recipes by muscle memory and could just mass produce stuff in 10 minutes, reaching the same maxed out RND issue as we have now.


Discussed on discord. With a complexity of 4, the complex chems are `ginsonic, fliptonium, degreaser, hair_dye, super_hairgrownium, initropidril, capulettium_plus, rotatium`, and that complexity can be increased. I do plan to have diminishing returns with each unit made though.

---

> Another hard cap/cooldown request for this, lest A) we get cargo bombarded with tank transfer valve requests, replacing the current "order muh prototypes rn" or B) the station will shake non-stop from potawater bombs for them points, greatly disrupting the flow of the round.


Again, hard caps, and youll only get decent points by hitting the target. I also have an idea to make an "explosion simulator" for TTVs, which can not only run the explosion without shaking the entire station, but also removes the doppler cheese (if you obtained multiple doppler arrays, all of them would trigger and add the point gain

---

> Once per extract, right? Xenobio can be spammed.


Diminishing returns, and it will be a far-bred slime which takes time to get to, which should also be a sufficient slowdown

---

> If chems/explosions give a flat point bonus, they will be spammed. Players want instant gratification, points over time is great, but it will be only relied on if they can overclock it to the point we get back to the level 7 by 20 minutes gameplay. I know this is the intention, but be ready for players trying to cheese it.


See above, mention of diminishing returns with each, as well as other stuff gone over

---

> This will greatly weaken science for nukie rounds (a good thing), but it will also inadvertently weaken cargo too (not sure if a good thing). Currently, during nukie rounds, RND gets done in about 8-10 minutes, ships the disks, cargo orders a lot of guns. If RND cannot get done anymore this fast, I suggest giving cargo a flat number of points from CC when war is declared to cut out science from this equation. Or a bump to RND levels from CC to make them able to do disks before the nukies arrive. Not enough to spam weapons yet but enough to get disks done/some equipment to be printed (I'm mainly thinking about medical stuff like cloner, sleeper, defibs, analyzers, handheld crew monitors...).


Obviously some levels will need to be tweaked around, or point gain will need to be raised or lowered. There is also the option of a "special case" cargo crate which can be enabled by admins, which supplies a circuit board for an R&D "super server" which could have a higher point gain, which is used on rounds where its total chaos and stuff like this is needed

---

> Without levels, I am unsure what robotics can do apart from cleanbots, cyborgs, and ripleys. Can they still do their fun projects or are they locked behind levels?


They have some base tier stuff they can do, and science will still research them anyways as time goes on, and again, levels can be tweaked.

---

This entire thing will take lots of testing, feedback, tweaking, repeating of that cycle, until its fully refined and playable. I dont plan to write it in a day and get it speedmerged full of glaring issues.


Any feedback is appreciated and I encourage you to find faults in the design.

**Comment by Denthamos** [comment]
As for strange objects, would it be better if they are moved from being randomly found on station to be RNG drop chance from ores on lavaland/space/gateway?  That way instead of a flat amount of strange objects found in maintenance the amount ebbs and flows depending on miners/explorers.

**Reply by AffectedArc07** [reply]
They could be space ruin finds, but that relies on having people go out and explore space, which is not only not super common, but it also encourages more off-station shenanigans, which I want to avoid. Ditto with the gateway.

**Reply by Denthamos** [reply]
Yeah, trying to avoid that aspect of having them spawn specifically in off space ruins and such.  Thats why I said ores specifically as those are not tied/locked to a single location and can be found by anyone at anytime if they happen to be destroying some rocks.  This would primarily tie in miners closer to the research gain as they will be loosing the aspect of giving them items to be researched from their vendor.

**Reply by AffectedArc07** [reply]
I think ores are a good idea, and it might make the system worthwhile.

My only issue with this approach is a ton of strange objects being put into crew circulation. 

I guess I could have the analyser "discover" it for a smaller amount of points, and "destroy" it for a higher amount of points.

**Reply by Denthamos** [reply]
Adding in more functions that the objects can do would also help in making them more unique and may lead to players wanting to keep it instead of destroy it due to the abilities of it.  I have a few ideas on this and will forward them to you.  Also on my end this is a 👍

**Comment by Necaladun** [comment]
There are concerns, nitpicks, and issues I have with this - mainly to do with my worry it will become a bit 'grindy' doing the same things.

However, idgaf. It looks better than what we have, and is a framework that allows for tweaking and changes. 👍

**Reply by AffectedArc07** [reply]
If you have any major ones to outline here please do. I will try address them as best I can as I want this to be an actual solution, not something that needs merging with 50 hotpatches to make it playable.

**Comment by DoctorDrugs** [comment]
I'm all for giving this a shot to see how things work out, I hope it's a better replacement for the current RND system as quite a lot of people have grown to hate it as it exists. I'm not a fan of techwebs myself, so seeing something else that has a design document and a starting point is great to see. 

Some things i'd like to see some thought put into a bit more is the research servers myself mostly regarding overclocking/upgrading them to generate more/faster points and keeping them cool. just a little something for optimizing and spending time working through ideas about how to best set these up would probably help with adding some longevity/complexity to it

**Comment by dearmochi** [comment]
Still rather rough for me to say, but I think it is a step into the right direction.

Echoing @DoctorDrugs, I would like a decent amount of depth to each feature to give people something to experiment for. Perhaps the research servers could have an assortment of parts/individual settings (clock speed, fan speed or w/e).

Another idea I had for gaining points is some kind of device that records nearby events. For example, holding it while somebody else makes a gargle blaster or while a doctor performs a surgery, it could grant some amount of points.

Going forward, the tycoon premise sounds good, now it would be good to add more substance and depth to your design doc

**Comment by marlyn-x86** [comment]
I do like the idea of tying research progress to some kind of point gain which can be scaled as necessary, and boosted by various actions that we want the players to perform

With how the research categories' benefits tend to not necessarily be restricted to a single department, I don't expect any issues where certain departments get consistently left in the dark or something, and if that happens we can just adjust point gain if necessary

Either way it seems like a good way for moving away from the current R&D shopping list meta and all its entangled expectations

**Comment by AffectedArc07** [comment]
Oh for those curious, this project isn't dead, just my motivation is.
